akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -514,44 +523,56 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
    *
    * All incoming messages are stashed.
    */
-  private def recovering(recoveryBehavior: Receive, previousReceiveTimeout: Duration) =
+  private def recovering(recoveryBehavior: Receive) =
     new State {
+
+      // protect against snapshot stalling forever because of journal overloaded and such
+      val timeout = extension.journalConfigFor(journalPluginId).getMillisDuration("recovery-event-timeout")
```

Comment:
You can't put it in the trait because that would break bin compat.
We could change `journalConfigFor` to return a `Settings` object that would read such things once,  but I don't think it's worth it. It's rather costly to start a persistent actor anyway.